FAILED tests/pytest/test_capture.py::test_healthcheck_traceback_is_hidden - assert 6 == 7PyPy3.8 7.3.6 test regression:

PyPy3.8 7.3.6 was released two days ago. With this version, I get the following new test failure:

```
============================================================== FAILURES ===============================================================
________________________________________________ test_healthcheck_traceback_is_hidden _________________________________________________
[gw2] linux -- Python 3.8.12 /usr/bin/pypy3
Traceback (most recent call last):
  File "/tmp/portage/dev-python/hypothesis-6.23.3/work/hypothesis-hypothesis-python-6.23.3/hypothesis-python-pypy3/tests/pytest/test_capture.py", line 110, in test_healthcheck_traceback_is_hidden
    assert timeout_line - def_line == expected
AssertionError: assert 6 == 7
  +6
  -7
-------------------------------------------------------- Captured stdout call ---------------------------------------------------------
running: /usr/bin/pypy3 -mpytest --basetemp=/tmp/portage/dev-python/hypothesis-6.23.3/temp/pytest-of-portage/pytest-0/popen-gw2/test_healthcheck_traceback_is_hidden0/runpytest-0 /tmp/portage/dev-python/hypothesis-6.23.3/temp/pytest-of-portage/pytest-0/popen-gw2/test_healthcheck_traceback_is_hidden0/test_healthcheck_traceback_is_hidden.py --verbose --basetemp=/tmp/portage/dev-python/hypothesis-6.23.3/temp/pytest-of-portage/pytest-0/popen-gw2/basetemp
     in: /tmp/portage/dev-python/hypothesis-6.23.3/temp/pytest-of-portage/pytest-0/popen-gw2/test_healthcheck_traceback_is_hidden0
========================================================= test session starts =========================================================
platform linux -- Python 3.8.12[pypy-7.3.6-final], pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/pypy3
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/tmp/portage/dev-python/hypothesis-6.23.3/temp/pytest-of-portage/pytest-0/popen-gw2/test_healthcheck_traceback_is_hidden0/.hypothesis/examples')
metadata: {'Python': '3.8.12', 'Platform': 'Linux-5.14.13-gentoo-dist-x86_64-AMD_Ryzen_5_3600_6-Core_Processor-with-glibc2.33', 'Packages': {'pytest': '6.2.5', 'py': '1.10.0', 'pluggy': '1.0.0'}, 'Plugins': {'hypothesis': '6.23.3', 'pyfakefs': '4.5.1', 'localserver': '0.5.0', 'case': '1.5.3', 'requests-mock': '1.9.3', 'betamax': '0.8.1', 'forked': '1.3.0', 'xprocess': '0.18.1', 'expect': '1.1.0', 'mock': '3.6.1', 'metadata': '1.11.0', 'subtests': '0.5.0', 'freezegun': '0.4.2', 'httpbin': '1.0.0', 'shutil': '1.7.0', 'virtualenv': '1.7.0', 'rerunfailures': '10.2', 'flaky': '3.7.0', 'aiohttp': '0.3.0', 'xdist': '2.4.0', 'cov': '3.0.0', 'timeout': '2.0.0', 'asyncio': '0.16.0'}, 'JAVA_HOME': '/etc/java-config-2/current-system-vm'}
rootdir: /tmp/portage/dev-python/hypothesis-6.23.3/temp/pytest-of-portage/pytest-0/popen-gw2/test_healthcheck_traceback_is_hidden0
plugins: hypothesis-6.23.3, pyfakefs-4.5.1, localserver-0.5.0, case-1.5.3, requests-mock-1.9.3, betamax-0.8.1, forked-1.3.0, xprocess-0.18.1, expect-1.1.0, mock-3.6.1, metadata-1.11.0, subtests-0.5.0, freezegun-0.4.2, httpbin-1.0.0, shutil-1.7.0, virtualenv-1.7.0, rerunfailures-10.2, flaky-3.7.0, aiohttp-0.3.0, xdist-2.4.0, cov-3.0.0, timeout-2.0.0, asyncio-0.16.0
collecting ... collected 1 item

test_healthcheck_traceback_is_hidden.py::test_healthcheck_traceback_is_hidden FAILED                                            [100%]

============================================================== FAILURES ===============================================================
________________________________________________ test_healthcheck_traceback_is_hidden _________________________________________________

>   ???
E   hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 5 valid examples in 1.01 seconds (0 invalid ones and 0 exceeded maximum size). Try decreasing size of the data you're generating (with e.g. max_size or max_leaves parameters).
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.too_slow to the suppress_health_check settings for this test.

test_healthcheck_traceback_is_hidden.py:5: FailedHealthCheck
------------------------------------------------------------- Hypothesis --------------------------------------------------------------
You can add @seed(253159079947732743015578611645523863276) to this test or run pytest with --hypothesis-seed=253159079947732743015578611645523863276 to reproduce this failure.
```

--- a/hypothesis-python/tests/pytest/test_capture.py
+++ b/hypothesis-python/tests/pytest/test_capture.py
@@ -17,7 +17,7 @@ import sys
 
 import pytest
 
-from hypothesis.internal.compat import WINDOWS, escape_unicode_characters
+from hypothesis.internal.compat import PYPY, WINDOWS, escape_unicode_characters
 
 pytest_plugins = "pytester"
 
@@ -106,7 +106,7 @@ def test_healthcheck_traceback_is_hidden(testdir):
     timeout_token = ": FailedHealthCheck"
     def_line = get_line_num(def_token, result)
     timeout_line = get_line_num(timeout_token, result)
-    expected = 6 if sys.version_info[:2] < (3, 8) else 7
+    expected = 6 if sys.version_info[:2] < (3, 8) or PYPY else 7
     assert timeout_line - def_line == expected
 
 

